client.go

```diff
@@ -83,7 +94,20 @@ func newClientTransport(c *clientSpec) *httptransport.Runtime {
 		return nil
 	})
 
-	transport := httptransport.New(c.host, c.basePath, c.schemes)
+	client := &http.Client{
+		Timeout: c.timeout,
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			Dial: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).Dial,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ResponseHeaderTimeout: 10 * time.Second,
+		},
```

Comment:
Maybe we could just use the Default Settings for the transport? It's a bit less code.

```suggestion
		Transport: http.DefaultTransport,
	}
```